### PR TITLE
SearchPortal Util and Module

### PR DESCRIPTION
**Pull request by prince-7**
Adding searchportal.co as util and module (fix for #136)
To Do:
- [x] adding to util/google

### CODE REVIEW

**Comment by saeeddhqan** [comment]
What does the https://cse.google.com/cse/element/v1 will do that google won't?
What is the difference between https://cse.google.com/cse/element/v1 and searchportal.co?

**Comment by prince-7** [comment]
> What does the https://cse.google.com/cse/element/v1 will do that google won't?

Similar results will be obtained, that's why we can use it as an alternative.
> What is the difference between https://cse.google.com/cse/element/v1 and searchportal.co?

The thing is that the search portal itself is requesting the results from  https://cse.google.com/cse/element/v1 the results are dynamically loaded so scraping the source code(of searchportal) won't give anything. so I directly requested from the google cse itself, and the captcha won't be triggered in this case.

**Comment by prince-7** [comment]
So this will cover all the search engines that use google enhancement.

**Comment by z3y50n** [comment]
This google custom search engine requires a token and has a limit I think. In this case `cx` is from your account and I don't think we should do that for an open source project where anyone can use this thing.

**Comment by prince-7** [comment]
It does need the token but we can use the token from the search engines that use gsce, what do you think?

**Comment by saeeddhqan** [comment]
Isn't it better to scrape the searchportal.co instead of ...?

**Comment by prince-7** [comment]
> Isn't it better to scrape the searchportal.co instead of ...?

Actually, the results are loaded on the dom dynamically so they can't be scraped, 
(eg. there is no results in the veiw-source code of the result page ->
view-source:https://www.google.com/search?num=10&start=0&ie=utf-8&oe=utf-8&q=owasp)
What to do?

**Comment by saeeddhqan** [comment]
> This google custom search engine requires a token and has a limit I think. In this case `cx` is from your account and I don't think we should do that for an open source project where anyone can use this thing.

It totally should have a limit, yes, but cx is stable for all! which means that it is specifically for the search portal. Also, I don't think that it could be useful for a captcha! but the output is very clean and I think we can have it.

**Comment by prince-7** [comment]
Please check this and merge.

**Comment by saeeddhqan** [comment]
The last step is to add this to the google util class as I told you in the slack.

**Comment by prince-7** [comment]
Done, Please check now.

File: core/util/searchportal.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+		else:
+			self._pages = req.text
+			try:
+				self._json = json.loads(req.text[34:-2])
+			except:
+				if req.status_code in (400,):
+					self.framework.error('The server cannot process the request because it is malformed')
+				self.framework.error('searchportal is missed!')
+				
+		return req
```
Comment (z3y50n) [inline core/util/searchportal.py:59]: If there is an exception will this `req` variable be undefined?

**Comment by saeeddhqan** [comment]
@prince-7 we need to crawl other pages!

**Comment by prince-7** [comment]
Changing the start would help to crawl different pages.

**Comment by saeeddhqan** [comment]
I'll complete this pr.

**Comment by saeeddhqan** [comment]
doesn't work

**Pull request closed without merging**